modules/storages/lib/api/v3/file_links/file_link_representer.rb

```diff
@@ -0,0 +1,136 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2022 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module API
+  module V3
+    module FileLinks
+      class FileLinkRepresenter < ::API::Decorators::Single
+        include API::Decorators::LinkedResource
+        include API::Decorators::DateProperty
+        include ::API::Caching::CachedRepresenter
+
+        property :id
+
+        date_time_property :created_at, writeable: false
+
+        date_time_property :updated_at, writeable: false
```

Comment:
The `writeable` properties are not required within our stack. In it the representers responsibility is to parse as much as possible. It is the contracts objective to check writability.

Reply:
Ok, I removed them.

Reply:
So that means that if there is an unexpected parameter, like `{ admin: true }` when saving a user, it is not silently discarded (like in strong_parameter: `params.permit(:this, :that)`). The contract detects it and returns an error.

Is it correct?

Reply:
That also means that representers are not meant to be used outside of a service, or if they are, they must be asociated with the right contract.

Reply:
done in 435c7855e3, but would still like some precisions.